58.md

```diff
@@ -0,0 +1,107 @@
+the badges (and order) to show on their profiles.
+
+The content is not used and SHOULD be blank.
+
+The `e` tags in the Profile Badge event MUST reference kind `8` events containing a `p` tag that matches that of the Profile Badge. Badges that don't meet this criteria MUST be ignored.
+
+### Motivation
+
+Users MAY be awarded badges in recognition, in gratitude, for participation, or in appreciation of a certain goal, task or cause.
+
+Users MAY choose to decorate their profiles with badges for fame, notoriety, recognition, support, etc., from badge issuers they deem reputable.
+
+### Recommendations
+
+Badge issuers MAY include some Proof of Work as per [NIP-13](13.md) when minting badges to embed them with an energy cost, arguably making them more special and valuable for users that wish to collect them.
+
+Clients MAY whitelist badge issuers (pubkeys) for the purpose of ensuring they retain a valuable/special factor for their users.
+
+Badge image recommended aspect ratio is 1:1 with a high-res size of 1024x1024 pixels.
+
+Badge thumbnail image recommended sizes are: 512x512 (xl), 256x256 (l), 64x64 (m), 32x32 (s) and 16x16 (xs).
+
+Clients SHOULD render the most appropriate badge thumbnail according to the number of badges chosen by the user and space available. Clients SHOULD render the high-res version on user action (click, tap, hover).
+
+### Example of Badge Event
+
+Event:
+```json
+{
+  "id": "<charlie's issued badge event id>",
+  "kind": 8,
+  "pubkey": "<charlie's (issuer) pubkey>",
+  "tags": [
+    ["p", "<alice's (recipient) pubkey in hex>"]
+  ],
+  "content": "<json stringified object shown below>",
+  ...other fields...
+}
+```
+
+Content:
+```json
+{
+  "long_desc": "This user joined our relay 2 years ago.",
+  "short_desc": "Two-Year Club",
+  "image": "https://relay.nostr/badges/two-year-club.png",
+  "thumbs": {
+    "512x512": "https://relay.nostr/badges/two-year-club-512x512.png",
+    "256x256": "https://relay.nostr/badges/two-year-club-256x256.png",
+    "128x128": "https://relay.nostr/badges/two-year-club-128x128.png",
+    "64x64": "https://relay.nostr/badges/two-year-club-64x64.png",
+    "32x32": "https://relay.nostr/badges/two-year-club-32x32.png",
+    "16x16": "https://relay.nostr/badges/two-year-club-16x16.png"
+  },
+  "sha256sums": {
+    "image": "8d43b3081c8091147e03e41878604606c8aa53d1337d49a4d5e0e8c29c223798",
+    "512x512": "af94c2afee5321b6acbca7ba00fcfa4ae2e103dda3a3a038e8b0aec4207ef164",
+    "256x256": "8c1eaf36d0353e20e30511c473411f1df0b9e58f68daadad972a8a6955824a0e",
+    "128x128": "86b0ebe1cf924f39a0eb4cfde5cdfe8a271b882284fc4f1c353bf6589eef1870",
+    "64x64": "a92ec141189c8e0228d8c388b9c0e822ecf1347367b834c43c7e82427569e06d",
+    "32x32": "6afa09c719f0729253ce8034618743d3a3aadd7aa4215554b7e5aa72b46c41a9",
+    "16x16": "abacf2675c05b7710cf853800a95e4191c10b2005540a39adb4b7cdda617249b"
+  }
```

Comment:
```suggestion
  "image": "https://relay.nostr/badges/two-year-club.png"
```

Reply:
Aren't these a little excessive? Who is going to check these checksums? It's annoying to do, probably can't be done in a browser. And also: if someone has actually included checksums, that means they are not intending to change their images -- but if they are intending to they can just _not include_ the checksums and get away with that, so I don't see how they're doing anything.

I think we shouldn't checksums now, but leave room for them to be specified later when the need arises.

About the thumbnails, I think they also should be optional.

Reply:
The checksums are _optional_ and there so someone can't change the image to a dick pic and deface your beautiful profile page.
If we can't verify this in web clients perhaps we should reconsider, but I thought it was possible just didn't look into it.

Reply:
> Aren't these a little excessive? Who is going to check these checksums? It's annoying to do, probably can't be done in a browser. And also: if someone has actually included checksums, that means they are not intending to change their images -- but if they are intending to they can just _not include_ the checksums and get away with that, so I don't see how they're doing anything.
> 
> 
> 
> I think we shouldn't checksums now, but leave room for them to be specified later when the need arises.
> 
> 
> 
> About the thumbnails, I think they also should be optional, and less sizes specified, there are so many here, not even Google requires so many different sizes for publishing apps to the Android store.

I can specify that the sizes there are only recommendations and are all optional. Would that suffice? The high resolution version of a badge may be expensive to download specially if clients allow you to set many on your profile, so we'll need thumbs.